samples/bpf/protect_gpt_user.c

```diff
@@ -0,0 +1,131 @@
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <linux/bpf.h>
+#include <errno.h>
+#include "trace_helpers.h"
+#include "bpf_load.h"
+
+/*
+ * user program to load bpf program (protect_gpt_kern) to prevent writing to GUID
+ * parititon table
+ *
+ * argument 1: device where program will be attached (ie. /dev/sda)
+ * argument 2: name for pinned program
+ * argument 3: --attach or --detach to attach/detach program
+*/
+
+int attach(char* dev, char* path);
+int detach(char* dev, char* path);
+
+int main(int argc, char **argv)
+{
+	char path[256];
+
+	if (argc != 4){
+		fprintf(stderr, "Error: invalid number of arguments\n");
```

Comment:
You can make this self documenting instead of the help in the documentation, like this:

usage(argv[0]);
return 1;

And usage can print something like this:

```
Usage:
  argv[0] <device> <name> --attach
  argv[0] <device> <name> --detach
```

Reply:
Updated, please review changes, not sure if they are exactly what you mean